lib/instrumentation/http-shared.js

```diff
@@ -79,14 +80,35 @@ exports.traceOutgoingRequest = function (agent, moduleName) {
   var ins = agent._instrumentation
 
   return function (orig) {
-    return function () {
+    return function (reqUrl, options, callback) {
       var span = agent.buildSpan()
       var id = span && span.transaction.id
 
       agent.logger.debug('intercepted call to %s.request %o', moduleName, { id: id })
 
-      var req = orig.apply(this, arguments)
-      if (!span) return req
+      if (!span) return orig.apply(this, arguments)
+
+      if (typeof options === 'function') {
+        callback = options
+        options = {}
+      }
+
+      if (typeof reqUrl === 'object' && !(reqUrl instanceof URL)) {
+        options = reqUrl
+      } else {
+        if (!(reqUrl instanceof URL)) {
+          reqUrl = new URL(reqUrl)
+        }
+        options = Object.assign({}, options, reqUrl)
+      }
+
+      if (!options.headers) options.headers = {}
+
+      // Use span context as traceparent header
+      options.headers['Elastic-APM-traceparent'] = span.context.toString()
```

Comment:
According to the design doc this needs to be `Elastic-APM-Traceparent` - it probably doesn't make a difference, but just in case

Reply:
Headers are technically _supposed_ to be case-insensitive, and Node.js _will_ actually lower-case it internally. I'm not sure how well-supported case-insensitivity of headers is across other platforms though. 🤔

Reply:
I agree it probably will not be an issue. I just mentioned it as it's technically says to use `Elastic-APM-Traceparent` in the design doc, so I thought it's better to just follow it by the letter.

When we remove our header prefix once TraceContext is finalized, the spec actually say to use `traceparent` (all lowercase) as there is certain issues with casing in the systems. So I guess sometimes it's important 🤷‍♂️

Reply:
fixed.